File: spring-reactive-02/learn-java/src/main/java/boki/learnjava/newsequence/range/RangeEx2.java
package boki.learnjava.newsequence.range;

import boki.learnjava.common.SampleData;
import boki.learnjava.utils.Logger;
import reactor.core.publisher.Flux;

import java.util.List;

/**
 * range() 사용 예제
 * 명령형 형식의 for 문을 대체하는 예제
 */
public class RangeEx2 {
    public static void main(String[] args) {
        List<String> coinNames = SampleData.coinNames;

        Flux
            .range(0, coinNames.size())
            .subscribe(index -> Logger.onNext(coinNames.get(index)));
    }
}

File: spring-reactive-02/learn-kt/src/main/kotlin/boki/learnkt/v1_newsequence/d_range/RangeEx2.kt
package boki.learnkt.v1_newsequence.d_range

import boki.learnjava.common.SampleData
import boki.learnkt.util.Logger
import reactor.core.publisher.Flux

/**
 * range() 사용 예제
 * 명령형 형식의 for 문을 대체하는 예제
 */
fun main() {
    val coinNames = SampleData.coinNames

    Flux
        .range(0, coinNames.size)
        .subscribe { Logger.onNext(coinNames[it]) }
}